### PR TITLE
Implement mock API server

### PR DESCRIPTION
**Pull request by alexhorn**
A mock server is needed to enable newcomers to develop without having to request an API key

### CODE REVIEW

**Comment by alexhorn** [comment]
https://mock.neuland.app/

File: mock-server/app.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
+const express = require('express')
+const bodyParser = require('body-parser')
+const cors = require('cors')
+const app = express()
+const port = 3001
+
+const SESSION_TOKEN = 'iamasessiontoken'
+
+const services = {
```
Comment (M4GNV5) [inline mock-server/app.js:9]: There are a lot of services missing

File: mock-server/app.js [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    session: {
+        open: params => ({
+            status: 0,
+            data: [
+                SESSION_TOKEN,
+                params.username,
+                0
+            ]
+        }),
+        isalive: params => ({
```
Comment (M4GNV5) [inline mock-server/app.js:19]: maybe we should put some minimal effort into these mock API requests (such as session&parameter checking).
This makes the mock server more useful for unit testing

File: mock-server/app.js [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    const method = req.body.method
+    const format = req.body.format
+
+    if (format !== 'json') {
+        res.status(400)
+    }
+
+    if (services.hasOwnProperty(service)) {
+        if (services[service].hasOwnProperty(method)) {
+            res.json(services[service][method](req.body))
```
Comment (M4GNV5) [inline mock-server/app.js:46]: We should probably wrap the service call into a try catch and mock the error status behaviour the official API responds with

**Review by M4GNV5 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**